mesonbuild/modules/fs.py

```diff
@@ -270,25 +271,38 @@ def read(self, state: 'ModuleState', args: T.Tuple['FileOrString'], kwargs: 'Rea
         return data
 
     @FeatureNew('fs.copyfile', '0.64.0')
-    @typed_pos_args('fs.copyfile', (File, str), optargs=[str])
+    @typed_pos_args('fs.copyfile', (File, str, build.GeneratedList, build.BuildTarget,
+                                    build.BothLibraries, build.CustomTargetIndex,
+                                    build.CustomTarget), optargs=[str])
     @typed_kwargs(
         'fs.copyfile',
         INSTALL_KW,
         INSTALL_MODE_KW,
         INSTALL_TAG_KW,
         KwargInfo('install_dir', (str, NoneType)),
     )
-    def copyfile(self, state: ModuleState, args: T.Tuple[FileOrString, T.Optional[str]],
+    def copyfile(self, state: ModuleState, args: T.Tuple[SourceInputs, T.Optional[str]],
                  kwargs: CopyKw) -> ModuleReturnValue:
         """Copy a file into the build directory at build time."""
         if kwargs['install'] and not kwargs['install_dir']:
             raise InvalidArguments('"install_dir" must be specified when "install" is true')
 
+        if isinstance(args[0], (build.GeneratedList, build.BuildTarget,
+                                build.BothLibraries, build.CustomTargetIndex,
+                                build.CustomTarget)):
+            FeatureNew.single_use('Copying from generated files', '1.3.0', state.subproject)
```

Comment:
Copying from build targets

Reply:
In fact, it can be more that build targets. I'm not sure how I should name it, both here and in the release notes...